docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -398,6 +398,43 @@ def downstream_asset(upstream):
 
 In this case, `ins={"upstream": AssetIn("upstream_asset")}` declares that the contents of the asset with the key `upstream_asset` will be provided to the function argument named `upstream`.
 
+### Attaching metadata
+
+Dagster supports attaching arbitrary metadata to assets. To attach arbitrary metadata, supply a `metadata` dictionary to the asset:
+
+```python file=/concepts/assets/asset_metadata.py startafter=start_simple_metadata endbefore=end_simple_metadata
+from dagster import asset
+
+@asset(
+    metadata={
+        "cereal_name": "Sugar Sprinkles"
+    },
+)
+```
+
+To specify data types for metadata values, use the <PyObject object="MetadataValue"/> class:
+
+```python file=/concepts/assets/asset_metadata.py startafter=start_typed_metadata endbefore=end_typed_metadata
+from dagster import asset, MetadataValue
+
+@asset(
+    metadata={
+        "cereal_name": "Sugar Sparkles",
+        "description": "Sparkly, sugary delight!",
```

Comment:
Just to keep things simple, I'd drop the `description` here since it doesn't add new info about metadata.